test/channels.jl

```diff
@@ -320,20 +321,17 @@ end
         @test !callback()
         @test timedwait(callback, 0) === :timed_out
 
-        @async begin sleep(0.5); put!(rr1, :ok) end
+        @async begin put!(sync, :ready); sleep(0.5); put!(rr1, :ok) end
         @async begin sleep(1.0); put!(rr2, :ok) end
-        @async begin sleep(2.0); put!(rr3, :ok) end
+        @async begin @test take!(rr3) == :done end
 
+        @test take!(sync) == :ready
         et = @elapsed timedwait(callback, 1)
 
-        # assuming that 0.5 seconds is a good enough buffer on a typical modern CPU
-        try
-            @assert (et >= 1.0) && (et <= 1.5)
-            @assert !isready(rr3)
-        catch
-            @warn "`timedwait` tests delayed. et=$et, isready(rr3)=$(isready(rr3))"
-        end
+        @test et >= 1.0
+
         @test isready(rr1)
```

Comment:
Why would `rr1` be ready here? That is still just assuming `sleep(0.5)` elapses before the `sleep(1)`?

Reply:
Yes, this is assuming that the dependency edge from the put! sync to the take! applies after the first sleep has enqueued its timer and that all timers have a global order. I think those are both true statements in the current system, but of course we may quibble on whether they are true semantically.